nsdb-core/src/main/scala/io/radicalbit/nsdb/index/FacetRangeIndex.scala

```diff
@@ -64,9 +64,10 @@ class FacetRangeIndex {
         throw new RuntimeException("Not implemented yet.")
     }
     toRecord(valueFieldType) {
-      facets.getTopChildren(0, rangeFieldName).labelValues.toSeq.map { lv =>
-        val structuredLabel = lv.label.split("-").map(_.toLong)
-        FacetRangeResult(structuredLabel(0), structuredLabel(1), lv.value)
+      facets.getTopChildren(0, rangeFieldName).labelValues.toSeq.collect {
+        case lv if lv.value.doubleValue > 0.0 =>
+          val structuredLabel = lv.label.split("-").map(_.toLong)
```

Comment:
is structuredLabel guaranteed to be always of two elements least?

Reply:
Yes, it is a Lucene structure. It's the class 

https://lucene.apache.org/core/8_0_0/facet/org/apache/lucene/facet/LabelAndValue.html

Unfortunately, there isn't exhaustive documentation. But looking at the source code unit tests,

https://github.com/apache/lucene-solr/blob/master/lucene/facet/src/test/org/apache/lucene/facet/range/TestRangeFacetCounts.java

we have inferred that the ranges returned are the same that are passed as input params.

Now... we have 2 long values that determine the range, but that Lucene class has only one string member for it.
So first we chain the 2 bounds and use the chained string as the lucene input param and then we split the result ranges in order to get our two long bounds back.

Hope it's clear enough... unfortunately it's tricky

Reply:
thank you very much